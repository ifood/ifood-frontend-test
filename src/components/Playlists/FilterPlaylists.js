import React from 'react'
import { Dropdown } from 'semantic-ui-react'

const returnOptions = (data) => {
    let options = [];
    data && data.length > 0 &&
    data.forEach(element => {
        let option = {
            key: element.value,
            text: element.name,
            value: element.value
        };
    options.push(option)    
    }); 
    return options;
}


const FilterPlaylists = ({playlistsFilters, handleChange}) => {
    const locale = playlistsFilters.filter((item) => item.id === 'locale')
    let values
    if(locale && locale.length > 0){
        values = returnOptions(locale[0].values)
    }
    const country = playlistsFilters.filter((item) => item.id === 'country')
    let countries
    if(country && country.length > 0){
        countries = returnOptions(country[0].values)
    }

    return (
        <div className='plyalist-filters'>
            <Dropdown
                placeholder='Select Locale'
                name='locale'
                fluid
                search
                selection
                className="filter-dropdown"
                onChange={handleChange}
                options={values || []}
            />

            <Dropdown
                placeholder='Select Country'
                selection
                search
                name='country'
                fluid
                className="filter-dropdown"
                options={countries || []}
                onChange={handleChange}
            />

            <input
                className='filter-limit'
                min='1'
                type='number'
                max="50"
                placeholder='Limit number'
                name='limit'
                onChange={handleChange}
            />

            <input
                className='filter-limit'
                type='date'
                name='timestamp'
                onChange={handleChange}
            />
            

            <input
                className='filter-limit'
                min={1}
                type='number'
                max={50}
                placeholder='Offset Number'
                name='offset'
                onChange={handleChange}
            />
        </div>
    )
}

export default FilterPlaylists
