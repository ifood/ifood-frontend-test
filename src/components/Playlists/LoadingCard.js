/*eslint-disable*/
import React from 'react';
import { LoadingContainer, Card } from './styles/LoadingCardStyles';

export default function LoadingCard() {
  return (
    <LoadingContainer>
      <Card variant="rect" width={320} height={400} />
      <Card variant="rect" width={320} height={400} />
      <Card variant="rect" width={320} height={400} />
      <Card variant="rect" width={320} height={400} />
      <Card variant="rect" width={320} height={400} />
      <Card variant="rect" width={320} height={400} />
      <Card variant="rect" width={320} height={400} />
      <Card variant="rect" width={320} height={400} />
      <Card variant="rect" width={320} height={400} />
      <Card variant="rect" width={320} height={400} />
    </LoadingContainer>
  );
}
