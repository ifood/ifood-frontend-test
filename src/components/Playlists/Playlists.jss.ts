import { makeStyles, Theme } from '@material-ui/core/styles';

export const useStyles = makeStyles(
  (theme: Theme) => ({
    container: {
      padding: theme.spacing(1),
    },
    item: {
      padding: theme.spacing(1),
    },
  }),
  { name: 'Playlists' }
);
