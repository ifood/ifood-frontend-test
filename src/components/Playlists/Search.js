import React from 'react'
import { Input } from 'semantic-ui-react'


const Search = ({ handleSearchChange }) => {
    return (
        <div className="search-input-container">
            <Input
                name='playlistName'
                icon='search'
                placeholder='Search...'
                className="search-input"
                onChange={handleSearchChange}
            />
        </div>

    )
}

export default Search
