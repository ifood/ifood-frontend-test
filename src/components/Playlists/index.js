/**
 *
 * Playlists
 *
 */

import React from 'react'
import PropTypes from 'prop-types'

import Loader from '../Loader'
import PlaylistCard from '../PlaylistCard'
import Text from '../Text'

import { PlaylistsWrapper } from './styles'

export function Playlists(props) {
  const {
    playlistResponse: {
      message,
      playlists,
    },
    nameFilter,
    isLoading,
  } = props

  if (isLoading) {
    return <Loader />
  }

  if (!playlists) {
    return null
  }

  const filteredPlaylists = playlists.items.reduce((acc, playlist) => {
    if (playlist.name.toUpperCase().includes(nameFilter.toUpperCase())) {
      acc.push(
        <PlaylistCard
          key={playlist.id}
          {...playlist}
        />,
      )
    }

    return acc
  }, [])

  return (
    <div>
      <Text bold uppercase big>
        {message}
      </Text>
      {filteredPlaylists.length ? (
        <PlaylistsWrapper>
          {filteredPlaylists}
        </PlaylistsWrapper>
      ) : (
        <Text>
          Nenhuma playlist encontrada
        </Text>
      )}
    </div>
  )
}

Playlists.propTypes = {
  playlistResponse: PropTypes.object.isRequired,
  nameFilter: PropTypes.string.isRequired,
  isLoading: PropTypes.bool,
}

export default Playlists
