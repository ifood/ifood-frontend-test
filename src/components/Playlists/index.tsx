import React from "react";

import { Container, Playlist } from "./styles";
import { IPlaylists } from "../../config/interfaces";

interface IPlaylistsProps {
  listItems: IPlaylists[];
}

const Playlists: React.FC<IPlaylistsProps> = ({ listItems }) => {
  return (
    <Container>
      {listItems.map((playlist) => (
        <Playlist
          key={playlist.id}
          className="appear-from-top"
        >
          <a href={playlist.external_urls.spotify} target="_blank" rel="noopener noreferrer" ><img src={playlist.images[0].url} alt={playlist.name} /></a>
          <div className="info">
            <a href={playlist.external_urls.spotify} target="_blank" rel="noopener noreferrer" ><strong>{playlist.name}</strong></a>
            <p>{playlist.description}</p>
            <div className="bottom">
              <span>by <a href={playlist.owner.external_urls.spotify} target="_blank" rel="noopener noreferrer">{playlist.owner.display_name}</a></span>
              <span>
                Colaborativa: {playlist.collaborative ? "Sim" : "NÃ£o"}
              </span>
              <span>{playlist.tracks.total} Faixas</span>
            </div>
          </div>
        </Playlist>
      ))}
    </Container>
  );
};

export default Playlists;
