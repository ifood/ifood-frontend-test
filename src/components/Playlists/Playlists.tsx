import React from 'react';
import Grid from '@material-ui/core/Grid';
import { PlaylistsItem } from './PlaylistsItem/PlaylistsItem';
import { Playlist } from '../../interfaces/playlist';
import { useStyles } from './Playlists.jss';

interface PlaylistsProps {
  playlists: Playlist[];
}

export const Playlists: React.FC<PlaylistsProps> = ({ playlists }) => {
  const classes = useStyles();
  return (
    <Grid className={classes.container} container role="list">
      {playlists.map((playlist) => (
        <Grid className={classes.item} item xs={6} sm={4} md={3} key={playlist.id} role="listitem">
          <PlaylistsItem playlist={playlist} />
        </Grid>
      ))}
    </Grid>
  );
};
