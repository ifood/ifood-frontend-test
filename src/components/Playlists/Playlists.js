/*eslint-disable*/
import React from 'react';
import usePlaylist from './hooks/usePlaylist';
import {
  PlaylistsContainer,
  PlaylistsCardContainer,
  PlaylistCard,
  PlaylistImg,
  PlaylistTextArea,
  Title,
  Header,
} from './styles/PlaylistsStyles';
import LoadingCard from './LoadingCard';
import SearchInput from './components/SearchInpunt';

export default function Playlists() {
  const { loading, title, playlists } = usePlaylist();
  const [filter, setFilter] = React.useState('');

  function hadleChange(ev) {
    setFilter(ev.target.value);
  }
  return (
    <PlaylistsContainer>
      <Header>
        <Title>{title}</Title>
        <SearchInput onChange={hadleChange} />
      </Header>
      {loading ? (
        <LoadingCard />
      ) : (
        <PlaylistsCardContainer>
          {playlists
            .filter(
              (item) =>
                item.name?.toLowerCase().indexOf(filter.toLowerCase()) !== -1
            )
            .map((item) => (
              <PlaylistCard key={item.id}>
                <PlaylistImg alt="playlist-img" src={item.images[0]?.url} />
                <PlaylistTextArea>
                  <span>{item.name}</span>
                  <span>{item.description}</span>
                </PlaylistTextArea>
              </PlaylistCard>
            ))}
        </PlaylistsCardContainer>
      )}
    </PlaylistsContainer>
  );
}
