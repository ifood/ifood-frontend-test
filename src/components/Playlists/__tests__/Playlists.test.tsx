import React from 'react';
import { render, screen } from '@testing-library/react';
import { Playlists } from '../Playlists';
import { AppProvider } from '../../../contexts/AppContext';

const playlists = [
  {
    id: '1',
    name: 'Playlist 1',
    description: 'Playlist 1',
    image: 'image1',
    url: 'url1',
  },
  {
    id: '2',
    name: 'Playlist 2',
    description: 'Playlist 2',
    image: 'image2',
    url: 'url2',
  },
  {
    id: '3',
    name: 'Playlist 3',
    description: 'Playlist 3',
    image: 'image3',
    url: 'url3',
  },
];

describe('Playlists', () => {
  test('matches the snapshot', () => {
    const { baseElement } = render(<Playlists playlists={playlists} />, { wrapper: AppProvider });
    expect(baseElement).toMatchSnapshot();
  });

  test('renders the playlists', () => {
    render(<Playlists playlists={playlists} />, { wrapper: AppProvider });

    expect(screen.getByRole('list')).toBeInTheDocument();
    expect(screen.getAllByRole('listitem')).toHaveLength(3);
  });
});
