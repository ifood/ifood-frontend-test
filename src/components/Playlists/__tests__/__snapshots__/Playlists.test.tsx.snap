// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Playlists matches the snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiGrid-root Playlists-container-1 MuiGrid-container"
      role="list"
    >
      <div
        class="MuiGrid-root Playlists-item-2 MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 MuiGrid-grid-md-3"
        role="listitem"
      >
        <div
          class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiCardActionArea-root"
            href="url1"
            tabindex="0"
          >
            <img
              class="MuiCardMedia-root PlaylistsItem-image-3 MuiCardMedia-media MuiCardMedia-img"
              src="image1"
              title="Imagem playlist"
            />
            <div
              class="MuiCardContent-root"
            >
              <p
                class="MuiTypography-root PlaylistsItem-name-4 MuiTypography-body1 MuiTypography-noWrap"
              >
                Playlist 1
              </p>
              <span
                class="MuiTypography-root PlaylistsItem-description-5 MuiTypography-caption MuiTypography-colorTextSecondary"
              >
                Playlist 1
              </span>
            </div>
            <span
              class="MuiCardActionArea-focusHighlight"
            />
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
      <div
        class="MuiGrid-root Playlists-item-2 MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 MuiGrid-grid-md-3"
        role="listitem"
      >
        <div
          class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiCardActionArea-root"
            href="url2"
            tabindex="0"
          >
            <img
              class="MuiCardMedia-root PlaylistsItem-image-3 MuiCardMedia-media MuiCardMedia-img"
              src="image2"
              title="Imagem playlist"
            />
            <div
              class="MuiCardContent-root"
            >
              <p
                class="MuiTypography-root PlaylistsItem-name-4 MuiTypography-body1 MuiTypography-noWrap"
              >
                Playlist 2
              </p>
              <span
                class="MuiTypography-root PlaylistsItem-description-5 MuiTypography-caption MuiTypography-colorTextSecondary"
              >
                Playlist 2
              </span>
            </div>
            <span
              class="MuiCardActionArea-focusHighlight"
            />
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
      <div
        class="MuiGrid-root Playlists-item-2 MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 MuiGrid-grid-md-3"
        role="listitem"
      >
        <div
          class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiCardActionArea-root"
            href="url3"
            tabindex="0"
          >
            <img
              class="MuiCardMedia-root PlaylistsItem-image-3 MuiCardMedia-media MuiCardMedia-img"
              src="image3"
              title="Imagem playlist"
            />
            <div
              class="MuiCardContent-root"
            >
              <p
                class="MuiTypography-root PlaylistsItem-name-4 MuiTypography-body1 MuiTypography-noWrap"
              >
                Playlist 3
              </p>
              <span
                class="MuiTypography-root PlaylistsItem-description-5 MuiTypography-caption MuiTypography-colorTextSecondary"
              >
                Playlist 3
              </span>
            </div>
            <span
              class="MuiCardActionArea-focusHighlight"
            />
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</body>
`;
